lib/pbench/server/cache_manager.py

```diff
@@ -1538,11 +1604,11 @@ def find_dataset(self, dataset_id: str) -> Tarball:
             dataset = Dataset.query(resource_id=dataset_id)
             sql = time.time()
             tarball = Path(Metadata.getvalue(dataset, Metadata.TARBALL_PATH))
-            controller_dir = tarball.parent
-            if controller_dir.name == dataset_id:
-                controller_dir = controller_dir.parent
         except Exception as e:
             raise TarballNotFound(dataset_id) from e
+        controller_dir = tarball.parent
+        if controller_dir.name == dataset_id:
+            controller_dir = controller_dir.parent
```

Comment:
Alternately,
```python
        tbp = tarball.parent
        controller_dir = tbp.parent if tbp.name == dataset_id else tbp
```
Ditto lines 1561-1563.

Reply:
Ugh. Ugly ternary alert ... ☹️

Reply:
Ugly is a bit judgey!  You want to assign one of two values to `controller_dir`, that just cries out for a ternary!!  (An `if` statement which overwrites `controller_dir`, now that's a candidate for "ugly"....  🙂)